force-app/main/default/classes/DuplicateSeasonHandler.cls

```diff
@@ -0,0 +1,47 @@
+public with sharing class DuplicateSeasonHandler {
+
+
+    public static void checkDuplicates(List<Season__c> seasons) {
+ 		
+        System.debug('###Season__c from trigger: ' + seasons);      
+        
+        List<Id> tvSerieIds = new List<Id>();
+        for(Season__c s : seasons) {
+            tvSerieIds.add(s.TV_Serie__c);
+        }
+        System.debug('###TV_Serie_Id from trigger: ' + tvSerieIds);
+        
+		Map<String,Id> mapSeasons = new Map<String,Id>();
+        // takes seasons that tv series id are in input values
+        List<Season__c> seasonsInDB = [SELECT Id,Sesaon_Name__c,TV_Serie__c FROM Season__c where TV_Serie__c IN :tvSerieIds];
+        
+        // create a map where key is a season name and value is tv serie
+        if(!seasonsInDB.isEmpty()) {
+        System.debug('###Taking seasons from DB');
+            for(Season__c season : seasonsInDB) {
+            mapSeasons.put(season.Sesaon_Name__c,season.TV_Serie__c);
```

Comment:
what if for 2 different tv series we will have seasons with the same name?
Create Map<String,List<Id>>
as String key: season name
as list<Id>: ids of tv series which are related with season with given name

Reply:
Done
The Map is now<Id,List>
Key : TV_Serie_Id
as list: Names of Series

TV_Serie_Id is always Unique that's why i set it as key ;)